data/schemas/keyboard.jsonschema

```diff
@@ -128,13 +235,118 @@
                 }
             }
         },
+        "split": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "enabled": {"type": "boolean"},
+                "matrix_grid": {
+                    "type": "array",
+                    "items": {"$ref": "qmk.definitions.v1#/mcu_pin"}
+                },
+                "matrix_pins": {
+                    "type": "object",
+                    "additionalProperties": false,
+                    "properties": {
+                        "right": {
+                            "type": "object",
+                            "additionalProperties": false,
+                            "properties": {
+                                "direct": {
+                                    "type": "array",
+                                    "items": {"$ref": "qmk.definitions.v1#/mcu_pin_array"}
+                                },
+                                "cols": {"$ref": "qmk.definitions.v1#/mcu_pin_array"},
+                                "rows": {"$ref": "qmk.definitions.v1#/mcu_pin_array"},
+                                "unused": {"$ref": "qmk.definitions.v1#/mcu_pin_array"}
+                            }
+                        }
+                    }
+                },
+                "main": {
+                    "type": "string",
+                    "enum": ["eeprom", "left", "matrix_grid", "pin", "right"]
+                },
+                "soft_serial_pin": {"$ref": "qmk.definitions.v1#/mcu_pin"},
+                "soft_serial_speed": {
+                    "type": "number",
+                    "min": 0,
+                    "max": 5,
+                    "multipleOf": 1
+                },
+                "transport": {
+                    "type": "object",
+                    "additionalProperties": false,
+                    "properties": {
+                        "protocol": {
+                            "type": "string",
+                            "enum": ["custom", "i2c", "serial"]
```

Comment:
Most of the time people will want to use `serial_usart` for split on ARM.
```suggestion
                            "enum": ["custom", "i2c", "serial", "serial_usart"]
```

The `serial_usart` driver has its own set of configurations:
* `#define SERIAL_USART_FULL_DUPLEX` (just `#ifdef`, no `TRUE`, default is undef'd)
* `#define SERIAL_USART_PIN_SWAP` (just `#ifdef`, no `TRUE`, default is undef'd)
* `#define SERIAL_USART_TX_PIN B6` - matches mcu pin
* `#define SERIAL_USART_RX_PIN B7` - matches mcu pin
* `#define SERIAL_USART_TX_PAL_MODE 7` - uint8
* `#define SERIAL_USART_RX_PAL_MODE 7` - uint8
* `#define SERIAL_USART_TIMEOUT 100` - uint8
* `#define SERIAL_USART_DRIVER SD1` - name of the serial driver, probably safe to limit between `SD1` and `SD9` -- unsure how many the biggest chips have but this should be safe for a first pass.

Reply:
That option wasn't documented on <https://docs.qmk.fm/#/config_options>. :/

How critical is it to add these other options? For this PR I was trying to limit it to everything found in `docs/config_options.md`.

Reply:
They are, on `develop`, using `qmk docs`:
![image](https://user-images.githubusercontent.com/2985843/129510174-21fa55cf-2e19-445e-b1cc-ece50582f7cc.png)